### PR TITLE
Add solution

### PR DESCRIPTION
**Pull request by OleksiiTromsa**

### CODE REVIEW

File: src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.persist(cinemaHall);
+            transaction.commit();
+            return cinemaHall;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Cannot insert CinemaHall entity", e);
```
Comment (dariiabutorina) [inline src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java]:
```suggestion
            throw new DataProcessingException("Cannot insert CinemaHall entity " + cinemaHall, e);
```

Comment (dariiabutorina) [reply src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java]: The message should be more informative for the developer to find the cause as fast as possible

File: src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.persist(movieSession);
+            transaction.commit();
+            return movieSession;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Cannot insert MovieSession entity", e);
```
Comment (dariiabutorina) [inline src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java]: Same

File: src/main/java/com/dev/cinema/model/CinemaHall.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+}
```
Comment (DariiaPikul) [inline src/main/java/com/dev/cinema/model/CinemaHall.java:56]: toString() - for every model (exceptions messages)

File: src/main/java/com/dev/cinema/service/impl/CinemaHallServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    public CinemaHall add(CinemaHall cinemaHall) {
+        return cinemaHallDao.add(cinemaHall);
+    }
+
+    @Override
+    public List<CinemaHall> getAll() {
+        return cinemaHallDao.getAll();
+    }
+
+}
```
Comment (dariiabutorina) [inline src/main/java/com/dev/cinema/service/impl/CinemaHallServiceImpl.java]:
```suggestion
    }
}
```

**Review by dariiabutorina — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<CinemaHall> getAll() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("from CinemaHall ",
+                    CinemaHall.class).getResultList();
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java]:
```suggestion
            return session.createQuery("from CinemaHall ", CinemaHall.class).getResultList();
```

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**